### PR TITLE
add is_valid function for dense model and fix example + unittest

### PR DESCRIPTION
**Pull request by fabinsch**
We had timeouts in debug mode on windows because the bounds were updated in the wrong order and it took too long to reach the maximum number of iterations.

### CODE REVIEW

**Review by jcarpent — APPROVED** [review]
Super nice fix. Thanks a lot @fabinsch

File: examples/cpp/solve_without_api.cpp [hunk 1]
```diff
@@ -11,13 +11,16 @@ using Vec = dense::Vec<T>;
 int
 main()
 {
+  ::proxsuite::proxqp::utils::rand::set_seed(1);
```
[the remark this answers is not in the document]
Comment (fabinsch) [reply examples/cpp/solve_without_api.cpp:14]: not really. I just wanted to have the same behavior for windows as for mac and linux.

Currently, not setting the seed at all leads to always the same seed in mac and linux, but to a random seed in windows. So this example is always the same for mac and linux, but it is always different on windows. 

I added this line here to have always the same example everywhere. But maybe I should rather work on the windows random number generator that by default it always has the same seed ? What do you think ?

[the remark this answers is not in the document]
Comment (jcarpent) [reply examples/cpp/solve_without_api.cpp:14]: as it is an example, it should be minimalist. So, if the feature is not important, we should remove it.

[the remark this answers is not in the document]
Comment (fabinsch) [reply examples/cpp/solve_without_api.cpp]: ok, I removed it here. And in general, I am setting a seed for windows (as it is done for macOS and linux currently).

File: include/proxsuite/linalg/veg/internal/macros.hpp [hunk 2]
```diff
@@ -24,6 +24,12 @@
     throw exception(ss.str());                                                 \
   }
 
+#define CHECK_DATA(size, expected_size)                                        \
```
Comment (jcarpent) [inline include/proxsuite/linalg/veg/internal/macros.hpp:27]: Should be renamed in PROXSUITE_CHECK_SIZE

Comment (jcarpent) [reply include/proxsuite/linalg/veg/internal/macros.hpp:27]: It would be nice to just use it locally as I don't think it is useful elsewhere

**Review by jcarpent — CHANGES_REQUESTED** [review]

**Review by jcarpent — APPROVED** [review]

**Pull request merged**